core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -300,11 +301,24 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
 
         CompletableFuture<Void> allFutures = CompletableFuture.allOf(
             futures.values().toArray(new CompletableFuture[0]));
-        return allFutures.thenApply(v -> {
+        return allFutures.handle((results, exception) -> {
```

Comment:
This is a CompletableFuture<Void> so `results` is not actually useful. Maybe `unused` would be better so this is clear.

Reply:
Done